dialog/form.go

```diff
@@ -0,0 +1,97 @@
+package dialog
+
+import (
+	"fyne.io/fyne"
+	"fyne.io/fyne/container"
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+)
+
+// formDialog is a simple dialog window for displaying FormItems inside a form.
+type formDialog struct {
+	*dialog
+	items   []*widget.FormItem
+	confirm *widget.Button
+	cancel  *widget.Button
+}
+
+// validateItems acts as a validation edge state handler that will respond to an individual widget's validation
+// state before checking all others to determine the net validation state. If the error passed is not nil, then the
+// confirm button will be disabled. If the error parameter is nil, then all other Validatable widgets in items are
+// checked as well to determine whether the confirm button should be disabled.
+// This method is passed to each Validatable widget's SetOnValidationChanged method in items by NewFormDialog.
+func (d *formDialog) validateItems(err error) {
+	if err != nil {
+		d.confirm.Disable()
+		return
+	}
+	for _, item := range d.items {
+		if validatable, ok := item.Widget.(fyne.Validatable); ok {
+			if err := validatable.Validate(); err != nil {
+				d.confirm.Disable()
+				return
+			}
+		}
+	}
+	d.confirm.Enable()
+}
+
+// NewFormDialog creates and returns a dialog over the specified application using
+// the provided FormItems. The cancel button will have the dismiss text set and the confirm button will
+// use the confirm text. The response callback is called on user action after validation passes.
+// If any Validatable widget reports that validation has failed, then the confirm
+// button will be disabled. The initial state of the confirm button will reflect the initial
+// validation state of the items added to the form dialog.
+func NewFormDialog(title, confirm, dismiss string, items []*widget.FormItem, callback func(bool),
+	parent fyne.Window) *formDialog {
```

Comment:
The tests are failing because golint is complaining about this being unexported. I think you should return fyne.Dialog instead, like @andydotxyz mentioned.

Reply:
Yeah, I just realized that wouldn't work at all. Sorry about that. Just pushed a fix.